### PR TITLE
Build docker image, etc.

### PR DESCRIPTION
**Pull request by syntrust**
Replaced https://github.com/ethstorage/es-node/pull/115

1. release zip with scripts and pre-built binaries
2. build a Docker image and push it to  GitHub Container Registry (ghcr.io) on each release
    - can be managed in https://github.com/orgs/ethstorage/packages
3. move the built binary (es-node) and snark folders(snarkjs, snarkbuild) to build/bin so that does not depend on src structures.

Note that the corresponding documentation will be changed in https://github.com/ethstorage/es-node/pull/92

### CODE REVIEW

File: run-rpc.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # usage:
 # ./run_rpc.sh
 
-executable="./cmd/es-node/es-node"
+executable="./es-node"
```
Comment (qzhodl) [inline run-rpc.sh:6]: Should change to build/bin, also need to update the rest content of this file

File: run.sh [hunk 2]
```diff
@@ -24,16 +24,15 @@ if [ ${#ES_NODE_SIGNER_PRIVATE_KEY} -ne 64 ]; then
...
 if [ ! -e  ${zkey_file} ]; then
-  echo "${zkey_file} not found. Start downloading..."
-  file_id="1ZLfhYeCXMnbk6wUiBADRAn1mZ8MI_zg-"
-  html=`curl -c ./cookie -s -L "https://drive.google.com/uc?export=download&id=${file_id}"`
-  curl -Lb ./cookie "https://drive.google.com/uc?export=download&`echo ${html}|grep -Po '(confirm=[a-zA-Z0-9\-_]+)'`&id=${file_id}" -o ${zkey_file}
-  echo "downloaded ${zkey_file}"
+echo "${zkey_file} not found, start downloading..."
+file_id="1ZLfhYeCXMnbk6wUiBADRAn1mZ8MI_zg-"
+html=`curl -c ./cookie -s -L "https://drive.google.com/uc?export=download&id=${file_id}"`
+curl -Lb ./cookie "https://drive.google.com/uc?export=download&`echo ${html}|grep -Po '(confirm=[a-zA-Z0-9\-_]+)'`&id=${file_id}" -o ${zkey_file}
```
Comment (qzhodl) [inline run.sh:32]: Need to update the option -P for macOS

File: run.sh [hunk 3]
```diff
@@ -24,16 +24,15 @@ if [ ${#ES_NODE_SIGNER_PRIVATE_KEY} -ne 64 ]; then
...
 # download blob_poseidon.zkey if not yet
-zkey_file="./ethstorage/prover/snarkjs/blob_poseidon.zkey"
+zkey_file="./build/bin/snarkjs/blob_poseidon.zkey"
 if [ ! -e  ${zkey_file} ]; then
-  echo "${zkey_file} not found. Start downloading..."
-  file_id="1ZLfhYeCXMnbk6wUiBADRAn1mZ8MI_zg-"
-  html=`curl -c ./cookie -s -L "https://drive.google.com/uc?export=download&id=${file_id}"`
-  curl -Lb ./cookie "https://drive.google.com/uc?export=download&`echo ${html}|grep -Po '(confirm=[a-zA-Z0-9\-_]+)'`&id=${file_id}" -o ${zkey_file}
-  echo "downloaded ${zkey_file}"
+echo "${zkey_file} not found, start downloading..."
```
Comment (qzhodl) [inline run.sh:29]: may need the indent

File: run.sh [hunk 4]
```diff
@@ -62,14 +61,14 @@ es_node_start=" --network devnet \
   --download.thread 32 \
   --p2p.max.request.size 4194304 \
   --p2p.max.concurrency 32 \
-  --p2p.bootnodes enr:-Li4QPFCNc7mLPqxoVrk1eKB0qa5hb8H75IBwhvdSGGdamx1egKibkKO1v1rtLt7r3pJvoVxv95ITlpSphYCAsunU6qGAYwkwuOpimV0aHN0b3JhZ2XbAYDY15S0tGvaqDX45LTY4gi2VZzSZ4UQUcGAgmlkgnY0gmlwhEFtcySJc2VjcDI1NmsxoQM9rkUZ7qWoJQT2UVrPzDRzmLqDrxCSR4zC4db-lgz1bYN0Y3CCJAaDdWRwgnZh \
```
Comment (qzhodl) [inline run.sh:65]: also need the indent

File: run.sh (hunk 2 above)
Comment (syntrust) [reply run.sh]: Tracked by https://github.com/ethstorage/es-node/issues/135

**Review by qzhodl — APPROVED** [review]

**Pull request merged**